src/ws/session.c

```diff
@@ -522,15 +522,22 @@ pam_conv_func_dummy (int num_msg,
 }
 
 static pam_handle_t *
-perform_tlscert (const char *rhost)
+perform_tlscert (const char *rhost,
+                 const char *authorization)
 {
   struct pam_conv conv = { pam_conv_func_dummy, };
+  const char *client_certificate;
   pam_handle_t *pamh;
   int res;
 
   debug ("start tls-cert authentication for cockpit-ws %u", getppid ());
 
-  char *username = cockpit_session_client_certificate_map_user ();
+  if (strncmp (authorization, "tls-cert ", 9) == 0)
```

Comment:
This is already done by `cockpit_authorize_type()`, so this could even be turned into an assertion, and a hard crash.